src/main/java/com/conveyal/analysis/components/HttpApi.java

```diff
@@ -79,14 +96,27 @@ public HttpApi (
             // Do not require authentication for internal API endpoints contacted by workers or for OPTIONS requests.
             String method = req.requestMethod();
             String pathInfo = req.pathInfo();
-            boolean authorize = pathInfo.startsWith("/api") && !"OPTIONS".equals(method);
+            boolean authorize = pathInfo.startsWith("/api") && !"OPTIONS".equalsIgnoreCase(method);
             if (authorize) {
-                // Determine which user is sending the request, and which permissions that user has. This method throws
-                // an error if the user cannot be authenticated.
-                UserPermissions user = authentication.authenticate(req);
-                LOG.info("{} {} by user {} in groups {}", method, pathInfo, user.email, user.groups);
+                // Determine which user is sending the request, and which permissions that user has.
+                // This method throws an exception if the user cannot be authenticated.
+                // Store the resulting permissions object in the request so it can be examined by any handler.
+                UserPermissions userPermissions = authentication.authenticate(req);
+                req.attribute(USER_PERMISSIONS_ATTRIBUTE, userPermissions);
+                // TODO stop using these two separate attributes, and use the permissions object directly
+                req.attribute(USER_EMAIL_ATTRIBUTE, userPermissions.email);
+                req.attribute(USER_GROUP_ATTRIBUTE, userPermissions.groups.iterator().next());
```

Comment:
I noticed the `events` collection in the database has a `groups` array, where every other collection has a single `accessGroup` string.

Unless we migrate everything else over we should revert `groups` being an array. Everywhere else we consider there to be a single `accessGroup`. We discussed a possible move to an array awhile back, but haven't had a need since we've been able to distinguish between administrators and everyone else.

Reply:
Yes, unless we have a specific plan to use multiple groups, we can switch it back to being a single string. It's a higher priority to make the database schema coherent across various tables.

Reply:
Marking this resolved since we've got a separate PR #648 to fix it.